### PR TITLE
use schema ids in apis

### PR DESCRIPTION
**Pull request by aramikm**
# Goal
The goal of this PR is address schema_id oriented graph and api structures

related to #35 

# Discussion
- Using schema ids instead of connection types
- add support for different environments

# Checklist
- [x] Fix Tests

### CODE REVIEW

File: config/src/lib.rs [hunk 1]
```diff
@@ -57,6 +61,48 @@ impl ConnectionType {
 	}
 }
 
+#[derive(Clone, PartialEq, Ord, Eq, PartialOrd, Debug)]
+pub enum Environment {
+	Mainnet,
+	Rococo,
+}
```
Comment (JoeCap08055) [inline config/src/lib.rs:68]: I think we also should have: `Dev(Config)` for ease of testing.

**Review by JoeCap08055 — COMMENTED** [review]
Looks mostly good. However, I don't think we need to be passing `Environment` down into all of the lower-level user/graph/page objects. Main use case is there will only ever be a single Environment for an application... can we make it a static at the crate level or something?

File: core/src/api/api.rs [hunk 2]
```diff
@@ -269,38 +270,40 @@ mod test {
...
-		let new_state = TestGraphState::<TEST_CAPACITY>::with_capacity(TEST_CAPACITY * 2);
+		let new_state =
+			TestGraphState::<TEST_CAPACITY>::with_capacity(Environment::Mainnet, TEST_CAPACITY * 2);
 		assert!(new_state.user_map.capacity() >= TEST_CAPACITY);
 	}
 
 	#[test]
 	fn graph_state_capacity() {
-		let state: TestGraphState = TestGraphState::new();
+		let state: TestGraphState = TestGraphState::new(Environment::Mainnet);
```
Comment (JoeCap08055) [inline core/src/api/api.rs:287]: lol, this scared me to reference "Mainnet" in a test suite, until I remembered that Mainnet config has nothing to do with communicating on-chain; it's just for constants & schema definition. Still scary to read when your eye first scans it, though 🤷🏻

**Review by JoeCap08055 — APPROVED** [review]

File: core/src/api/api.rs (hunk 2 above)
Comment (aramikm) [reply core/src/api/api.rs:287]: Yeah we would need some more tests with some non mainnet configs to make sure everything works as expected but will do that in another PR.

**Pull request merged**